test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.client.cs

```diff
@@ -480,37 +477,21 @@ public void GrpcPropagatesContextWithSuppressInstrumentationOptionSetToTrue()
+                .AddInMemoryExporter(exportedItems)
                 .Build())
             {
-                using (var activity = source.StartActivity("parent"))
-                {
-                    Assert.NotNull(activity);
-                    var channel = GrpcChannel.ForAddress(uri);
-                    var client = new Greeter.GreeterClient(channel);
-                    var rs = client.SayHello(new HelloRequest());
-                }
-
-                WaitForProcessorInvocations(processor, 7);
+                using var activity = source.StartActivity("parent");
+                Assert.NotNull(activity);
+                var channel = GrpcChannel.ForAddress(uri);
+                var client = new Greeter.GreeterClient(channel);
+                var rs = client.SayHello(new HelloRequest());
             }
 
-            Assert.Equal(9, processor.Invocations.Count); // SetParentProvider + (OnStart + OnEnd) * 3 (parent, gRPC client, and server) + Shutdown + Dispose called.
-
-            Assert.Single(processor.Invocations, invo => invo.Method.Name == "SetParentProvider");
-            Assert.Single(processor.Invocations, GeneratePredicateForMoqProcessorActivity(nameof(processor.Object.OnStart), "parent"));
-            Assert.Single(processor.Invocations, GeneratePredicateForMoqProcessorActivity(nameof(processor.Object.OnStart), OperationNameGrpcOut));
-            Assert.Single(processor.Invocations, GeneratePredicateForMoqProcessorActivity(nameof(processor.Object.OnStart), OperationNameHttpRequestIn));
-            Assert.Single(processor.Invocations, GeneratePredicateForMoqProcessorActivity(nameof(processor.Object.OnEnd), OperationNameHttpRequestIn));
-            Assert.Single(processor.Invocations, GeneratePredicateForMoqProcessorActivity(nameof(processor.Object.OnEnd), OperationNameGrpcOut));
-            Assert.Single(processor.Invocations, GeneratePredicateForMoqProcessorActivity(nameof(processor.Object.OnEnd), "parent"));
-            Assert.Single(processor.Invocations, invo => invo.Method.Name == "OnShutdown");
-            Assert.Single(processor.Invocations, invo => invo.Method.Name == nameof(processor.Object.Dispose));
-
-            var serverActivity = GetActivityFromProcessorInvocation(processor, nameof(processor.Object.OnEnd), OperationNameHttpRequestIn);
-            var clientActivity = GetActivityFromProcessorInvocation(processor, nameof(processor.Object.OnEnd), OperationNameGrpcOut);
+            var serverActivity = exportedItems.Single(activity => activity.OperationName == OperationNameHttpRequestIn);
+            var clientActivity = exportedItems.Single(activity => activity.OperationName == OperationNameGrpcOut);
 
             Assert.Equal($"greet.Greeter/SayHello", clientActivity.DisplayName);
-            Assert.Equal($"greet.Greeter/SayHello", serverActivity.DisplayName);
+            Assert.Equal($"POST /greet.Greeter/SayHello", serverActivity.DisplayName);
```

Comment:
curious how the expected value is changed now?

Reply:
ah, this test is skipped. It is related to https://github.com/open-telemetry/opentelemetry-dotnet/pull/5026